xarray/plot/dataset_plot.py

```diff
@@ -299,61 +309,176 @@ def newplotfunc(
 
         return primitive
 
-    @functools.wraps(newplotfunc)
-    def plotmethod(
-        _PlotMethods_obj,
-        x=None,
-        y=None,
-        u=None,
-        v=None,
-        hue=None,
-        hue_style=None,
-        col=None,
-        row=None,
-        ax=None,
-        figsize=None,
-        col_wrap=None,
-        sharex=True,
-        sharey=True,
-        aspect=None,
-        size=None,
-        subplot_kws=None,
-        add_guide=None,
-        cbar_kwargs=None,
-        cbar_ax=None,
-        vmin=None,
-        vmax=None,
-        norm=None,
-        infer_intervals=None,
-        center=None,
-        levels=None,
-        robust=None,
-        colors=None,
-        extend=None,
-        cmap=None,
-        **kwargs,
-    ):
-        """
-        The method should have the same signature as the function.
-
-        This just makes the method work on Plotmethods objects,
-        and passes all the other arguments straight through.
-        """
-        allargs = locals()
-        allargs["ds"] = _PlotMethods_obj._ds
-        allargs.update(kwargs)
-        for arg in ["_PlotMethods_obj", "newplotfunc", "kwargs"]:
-            del allargs[arg]
-        return newplotfunc(**allargs)
-
-    # Add to class _PlotMethods
-    setattr(_Dataset_PlotMethods, plotmethod.__name__, plotmethod)
+    # we want to actually expose the signature of newplotfunc
+    # and not the copied **kwargs from the plotfunc which
+    # functools.wraps adds, so delete the wrapped attr
+    del newplotfunc.__wrapped__
 
     return newplotfunc
 
 
+@overload
+def scatter(
+    ds: Dataset,
+    x: Hashable | None = None,
+    y: Hashable | None = None,
+    u: Hashable | None = None,
+    v: Hashable | None = None,
+    *,
```

Comment:
```suggestion
    *args: Any,
    x: Hashable | None = None,
    y: Hashable | None = None,
    u: Hashable | None = None,
    v: Hashable | None = None,
```

I'm not convinced these should be positional arguments.

We have like 3 major "types" of plotting functions: 
* `plot.line` - x, y are keyword only, has *args for the classical mpl formatting arguments: `'.-'`
* `_2dplot` - u, v are not used, yet
* `_dsplot`.  - has x,y,u,v but no *args, probably because line-plots were never implemented.

My intention is to add a `line` plot for datasets (pretty much working here #5622), so then we will likely need to add in `*args`.  `quiver` will likely fit in `_2dplot` eventually, then u,v has to be added there. Once we have more support for 3d plots (scatter and line examples here #5622), then it would feel intuitive to have this kind of signature:
```python
    *args: Any,
    x: Hashable | None = None,
    y: Hashable | None = None,
    z: Hashable | None = None,
    ...
```

It would be a complete mess trying to keep the backwards compatibility with positional and keyword arguments. If we use keywords only we can pretty much use the same signature on all the functions as well, which surely must make the maintenance of this much easier.

Reply:
Ok, I can change that.
What should *args do in the meantime? Just nothing?

Reply:
Ok another problem:
when we add `*args` to all plotting functions then calling it with positional `x` and `y` will silently fail. How do we prevent that? For now check that *args is empty?

Reply:
yeah, just nothing I was thinking.